### PR TITLE
Feature/catx a1 sldt@a1 sldt 178 triplestore

### PR DESCRIPTION
**Pull request by fmisir**
Replaces the RDBMS storage with a TripleStore.

The API do no longer have the fields:
   - private
   - publisher
   - the sate DEPRECATED as we do not have any implementation for it yet

The other changes in the API definition are name changes of the "Model" object. The name "Model" conflicts with the Jena Model class, full imports were making the code unreadable.

All tests are adjusted. Currently the Filter API is not finalized, I'am working on it, this PR can be reviewed meanwhile. There will be no more big changes.

### CODE REVIEW

**Review by JaneMcBrain — APPROVED** [review]

File: semantics/services/src/main/java/net/catenax/semantics/hub/domain/ModelPackageStatus.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.catenax.semantics.hub.domain;
+
+public enum ModelPackageStatus {
+   RELEASED,
+   DRAFT
```
Comment (drcgjung) [inline semantics/services/src/main/java/net/catenax/semantics/hub/domain/ModelPackageStatus.java:20]: DEPRECATED!

File: semantics/services/src/main/java/net/catenax/semantics/hub/persistence/triplestore/TripleStorePersistence.java [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+import net.catenax.semantics.hub.persistence.PersistenceLayer;
+
+public class TripleStorePersistence implements PersistenceLayer {
+
+   private final RDFConnectionRemoteBuilder rdfConnectionRemoteBuilder;
+   private final SdsSdk sdsSdk;
+
+   public TripleStorePersistence( final RDFConnectionRemoteBuilder rdfConnectionRemoteBuilder,
+         final SdsSdk sdsSdk ) {
+      this.rdfConnectionRemoteBuilder = rdfConnectionRemoteBuilder;
```
Comment (drcgjung) [inline semantics/services/src/main/java/net/catenax/semantics/hub/persistence/triplestore/TripleStorePersistence.java:60]: So you are already using a remote connection to the embedded fuseki?

File: semantics/services/src/main/resources/application.yml [hunk 3]
```diff
@@ -29,6 +29,21 @@ server:
     key-store-type: PKCS12
     key-alias: 1
     enabled: false
+
+hub:
+  triple-store:
+    embedded:
```
Comment (drcgjung) [inline semantics/services/src/main/resources/application.yml:35]: how would using an external SPARQL/http server be configured?

**Review by drcgjung — APPROVED** [review]
Two main points before approving to main:
- make embedded Fuseki optional (<scope>test</scope>) and only in the services module
- (re-)add DEPRECATED status.

Otherwise:
- great work!

**Pull request merged**